app/src/main/java/com/graddex/graddex/PokemonAPIFetch.kt

```diff
@@ -0,0 +1,57 @@
+package com.graddex.graddex
+
+import android.app.Application
+import android.nfc.Tag
+import android.util.Log
+import android.view.View
+import com.chuckerteam.chucker.api.ChuckerInterceptor
+import com.graddex.graddex.models.PokemonResponse
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.*
+import java.io.File
+import java.io.IOException
+
+class PokemonAPIFetch {
+    fun fetch(application: Application, tag: String, onSuccess: (res: PokemonResponse) -> Unit, onFailure: () -> Unit){
+        // Initialise rest client with cache and implement a JSON adapter
+        val cache = Cache(
+            File(application.cacheDir, "http_cache"),
+            50L * 1024L * 1024L)
+        val client = OkHttpClient.Builder()
+            .cache(cache)
+            .addInterceptor(ChuckerInterceptor(application))
+            .build()
+        val moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
+        val adapter: JsonAdapter<PokemonResponse> = moshi.adapter(PokemonResponse::class.java)
```

Comment:
maybe we could put these outside the function, as class properties, since they can be reused as `fetch` is called